ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/methods/engine/EngineNewPayload.java

```diff
@@ -187,17 +187,8 @@ public JsonRpcResponse syncResponse(final JsonRpcRequestContext requestContext)
           mergeContext.get().isSyncing(),
           parentHeader.isEmpty(),
           block.getHash());
-      mergeCoordinator
-          .appendNewPayloadToSync(block)
-          .exceptionally(
-              exception -> {
-                debugLambda(
-                    LOG,
-                    "Sync to block {} failed, reason {}",
-                    block::toLogString,
-                    exception::getMessage);
-                return null;
-              });
+      mergeCoordinator.appendNewPayloadToSync(block);
```

Comment:
what's the impact of removing this exceptionally clause?

Reply:
A big improvement in terms of reducing java heap, since was keeping reference to blocks sent by newPayload, that causes high memory consumption during initial sync, and could be one of the causes that prevent to complete snap sync on low spec machines.
Exceptions are also logged by the backward sync, so there is no loss of information.

Reply:
👍